src/gradata/hooks/inject_brain_rules.py

```
@@ -0,0 +1,61 @@
+
+    text = lessons_path.read_text(encoding="utf-8")
+    all_lessons = parse_lessons(text)
+    filtered = [l for l in all_lessons if l.state.name in ("RULE", "PATTERN") and l.confidence >= MIN_CONFIDENCE]
+    if not filtered:
+        return None
+
+    scored = sorted(filtered, key=_score, reverse=True)[:MAX_RULES]
+
+    lines = []
+    for r in scored:
+        lines.append(f"[{r.state.name}:{r.confidence:.2f}] {r.category}: {r.description}")
```

Ruff (0.15.9)

[warning] Use a list comprehension to create a transformed list

(PERF401)